### PR TITLE
RUN-671: Code changes for transferred Ansible Repo

### PR DESCRIPTION
**Pull request by qualman**
### To be released as 3.2.0

* Use groovy instead of Java
* Change build.gradle to publish to our artifact repo
* Publish as a GitHub Release via GitHub Actions, JitPack picks it up
* Change module/package names, folder paths
* Hook up to build pipeline, remove .travis.yml
* Edit README
* Dockerfile metadata / project.properties

A release with the following PRs would adjust Rundeck/Rundeckpro to point at this new location and version
* https://github.com/rundeck/rundeck/pull/7607
* https://github.com/rundeckpro/rundeckpro/pull/2417

### Testing:
To test, either build this as a Jar and put in `libext` for your local Rundeckpro, or move the whole folder into `rundeckpro/plugins`, and change build files to include it

### CODE REVIEW

File: docker/project.properties [hunk 1]
```diff
@@ -8,9 +8,9 @@ resources.source.1.config.ansible-become=false
 resources.source.1.config.ansible-gather-facts=true
 resources.source.1.config.ansible-ignore-errors=true
 resources.source.1.config.ansible-inventory=/home/rundeck/data/inventory.ini
-resources.source.1.type=com.batix.rundeck.plugins.AnsibleResourceModelSourceFactory
-service.FileCopier.default.provider=com.batix.rundeck.plugins.AnsibleFileCopier
-service.NodeExecutor.default.provider=com.batix.rundeck.plugins.AnsibleNodeExecutor
+resources.source.1.type=com.rundeck.plugins.ansible.plugin.AnsibleResourceModelSourceFactory
+service.FileCopier.default.provider=com.rundeck.plugins.ansible.plugin.AnsibleFileCopier
+service.NodeExecutor.default.provider=com.rundeck.plugins.ansible.plugin.AnsibleNodeExecutor
```
Comment (ltamaster) [inline docker/project.properties]: this hasn't changed, the name is set here 
https://github.com/rundeck-plugins/ansible-plugin/blob/enh/RUN-671_migrate_repo-220310/src/main/groovy/com/rundeck/plugins/ansible/plugin/AnsibleResourceModelSourceFactory.java#L17

https://github.com/rundeck-plugins/ansible-plugin/blob/enh/RUN-671_migrate_repo-220310/src/main/groovy/com/rundeck/plugins/ansible/plugin/AnsibleFileCopier.java#L27

https://github.com/rundeck-plugins/ansible-plugin/blob/enh/RUN-671_migrate_repo-220310/src/main/groovy/com/rundeck/plugins/ansible/plugin/AnsibleNodeExecutor.java#L25

**Review by ltamaster — CHANGES_REQUESTED** [review]

File: docker/project.properties (hunk 1 above)
Comment (qualman) [reply docker/project.properties]: I am not sure how my find and replaces missed this.... thank you!

**Comment by ltamaster** [comment]
now I am thinking that we shouldn't change the provider name because of this
Screen Shot 2022-03-30 at 13 39 48

it will break existing projects

**Comment by ltamaster** [comment]
well for resource-model, node-executor and file-copier it is easy to migrate (just replace project.properties with the new name) 
```
resources.source.2.type=com.rundeck.plugins.ansible.plugin.AnsibleResourceModelSourceFactory
service.FileCopier.default.provider=com.rundeck.plugins.ansible.plugin.AnsibleFileCopier
service.NodeExecutor.default.provider=com.rundeck.plugins.ansible.plugin.AnsibleNodeExecutor
```

but for jobs, that will be harder because you will need to export them, change the provider names and import them:

Screen Shot 2022-03-30 at 14 52 09

Screen Shot 2022-03-30 at 14 52 16

**Comment by qualman** [comment]
Ok @ltamaster , see if this latest change makes those jobs work for you?

**Review by ltamaster — APPROVED** [review]
tested

**Pull request merged**